ego4d/internal/university_files.py

```diff
@@ -0,0 +1,721 @@
+    annotations_dict: Dict[str, Annotations],
+    devices: Dict[str, Device],
+    component_types: Dict[str, ComponentType],
+    scenarios: Dict[str, Scenario],
+    bucket_name: str,
+    s3
+) -> None:
+    error_message = ""
+    # Check ids in video_components_dict are in video_metadata_dict
+    # and the # of components is correct
+    s3_bucket = S3Helper(s3, bucket_name)
+    video_info_param = []
+
+    for video_id, components in video_components_dict.items():
+        if video_id not in video_metadata_dict:
+            error_message += (
+                "video_component_file points to video_metadata at"
+                f" non-existent university_video_id: '{video_id}'\n"
+            )
+        elif video_metadata_dict[video_id].number_video_components != len(components):
+            error_message += (
+                f"video_metadata: '{video_id}'has {len(components)} components when it"
+                f" should have {video_metadata_dict[video_id].number_video_components}\n"
+            )
+        components.sort(key=lambda x: x.component_index)
+        #check component_index is incremental and starts at 0
+        university_video_folder_path = video_metadata_dict[video_id].university_video_folder_path
+        for i in range(len(components)):
+            if components[i].component_index != i:
+                error_message += (
+                f"video_metadata: '{video_id}' has component index {components[i].component_index}"
+                f" when it should have {i}\n"
+            )
+
+            #check component S3 files
+            s3_path = f"{university_video_folder_path.rstrip('/')}/{components[i].video_component_relative_path}"
+            bucket, key = split_s3_path(s3_path)
+            if bucket != bucket_name:
+                error_message += (
+                f"video_metadata: '{video_id}' has wrong bucket name"
+            )
+            # if not s3_bucket.exist(bucket, key):
+            #     error_message += (
+            #     f"video_component: '{s3_path}' does not exist in bucket"
+            # )
+            video_info_param.append((video_id, key))
+
+    #   check all mp4 data
+    video_info_dict = defaultdict(list)
+
+    def thread_helper(x):
+        video_info = get_video_info(s3, bucket_name, x[1])
+        lock.acquire()
+        video_info_dict[x[0]].append(video_info)
+        lock.release()
+
+    with ThreadPoolExecutor(max_workers=15) as pool:
+        tqdm(
+            pool.map(
+                    thread_helper,
+                    video_info_param
+                ),
+            total = len(video_info_param)
+        )
+
+    for video_id, video_infos in video_info_dict.items():
+        total_codec = set()
+        total_dimension = set()
+        for video_info in video_infos:
+            if video_info.codec != None:
+                total_codec.add(video_info.codec)
+                if len(total_codec) > 1:
+                    logging.error(
```

Comment:
similar to my other comment, rather than log let's return a list of errors and/or warnings